website/content/en/docs/best-practices/resource-pruning.md

```diff
@@ -0,0 +1,149 @@
+| Strategy.CustomSettings| a golang map of values that can be passed into a Custom strategy function
+| PreDeleteHook| optionally specifies a golang function to call before pruning a resource
+| CustomStrategy | optionally specifies a golang function that implements a custom pruning strategy
+
+
+### Pruning Execution
+
+Users can invoke the pruning by running the *Execute* function on the pruning configuration
+as follows:
+```golang
+err := cfg.Execute(ctx)
+```
+
+Users might want to implement pruning execution by means of a cron package or simply call the prune
+library based on some other triggering event.
+
+## Pruning Strategies
+
+### maxcount Strategy
+
+A strategy of leaving a finite set of resources is implemented called *maxcount*.  This strategy
+seeks to leave a specific number of resources, sorted by latest, on your cluster.  For example if
+you have 10 resources that would be pruned, and you specified a *maxcount* value of 4, then 6 
+resources would be pruned (removed) from your cluster starting with the oldest resources.
+
+### maxage Strategy
+
+A strategy of removing resources greater than a specific time is called *maxage*.  This strategy
+seeks to remove resources older than a specified *maxage* duration.  For example, a library
+user might specify a value of *48h* to indicate that any resource older than 48 hours would be
+pruned.  Durations are specified using golang's time.Duration formatting (e.g. 48h).
+
+## Pruning Customization
+
+### preDelete Hook
+
+Users can provide a *preDelete* hook when using the [operator-lib prune package][operator-lib-prune].  
+This hook function will be called by the library before removing a resource.  This provides a means to examine
+the resource logs for example, extracting any valued content, before the resource is removed
+from the cluster.
+
+Here is an example of a *preDelete* hook:
+```golang
+func myhook(cfg Config, x ResourceInfo) error {
+	fmt.Println("myhook is called ")
+	return nil
+}
+```
+
+Note that if your custom hook returns an error, then the resource will not be removed by the
+prune library.
+
+### Custom Strategy
+
+Library users can also write their own custom pruning strategy function to support advanced
+cases.  Custom strategy functions are passed in the prune configuration and a list of resources selected by
+the library.  Here is an example custom strategy:
+```golang
+func myStrategy(cfg Config, resources []ResourceInfo) (resourcesToRemove []ResourceInfo, err error) {
+	fmt.Printf("myStrategy is called with resources %v config %v\n", resources, cfg)
+	if len(resources) != 3 {
+		return resourcesToRemove, fmt.Errorf("count of resources did not equal our expectation")
+	}
+	return resourcesToRemove, nil
+}
+```
+
+A custom strategy builds up a list of resources to be removed, returning the list to the prune library which
+performs the actual resource removal. 
```

Comment:
I moved this up to the top of this section so that it flows better, at least IMO.

```suggestion
```